specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -12605,17 +12719,88 @@
           "$ref": "#/definitions/WindowsConfiguration",
           "title": "Windows operating system settings on the virtual machine.",
           "description": "This property must not be specified if the imageReference or osDisk property specifies a Linux OS image."
+        },
+        "dataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataDisk"
+          },
+          "title": "The configuration for data disks attached to the comptue nodes in the pool.",
+          "description": "This property must be specified if the compute nodes in the pool need to have empty data disks attached to them. This cannot be updated."
+        },
+        "licenseType": {
+          "type": "string",
+          "title": "If the the image or disk that is being used was licensed on-premises.",
+          "description": "This only applies to images that contain the Windows operating system. Values are:\n\n Windows_Server - The on-premises license is for Windows Server.\n Windows_Client - The on-premises license is for Windows Client."
+        },
+        "containerConfiguration": {
+          "$ref": "#/definitions/ContainerConfiguration",
+          "title": "The container configuration for the pool.",
+          "description": "If this is specified, then all regular tasks and job manager tasks run on this pool must specify the containerSettings property."
         }
       },
-      "required": [ "nodeAgentSKUId" ],
+      "required": [ "nodeAgentSKUId", "imageReference" ],
       "title": "The configuration for compute nodes in a pool based on the Azure Virtual Machines infrastructure."
     },
+    "ContainerRegistry": {
+      "properties": {
+        "registryServer": {
+          "type": "string",
+          "title": "The registry server url.",
+          "description": "If omitted, the default is: \"docker.io\"."
+        },
+        "username": {
+          "type": "string",
+          "x-ms-client-name": "userName",
+          "title": "The user name to login to the registry server."
+        },
+        "password": {
+          "type": "string",
+          "title": "The password to login to the registry server."
+        }
+      },
+      "required": [ "username", "password" ],
+      "title": "Private container registry details."
+    },
+    "ContainerConfiguration": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "Which container technology will be used.",
```

Comment:
minor style nit: "The container technology to be used." perhaps?  I think we generally use "The" rather than "Which" for these things but could be wrong.

Reply:
Fixed